--- a/node_modules/react-native/React.podspec
+++ b/node_modules/react-native/React.podspec
@@ -115,6 +115,8 @@
   s.subspec "jsiexecutor" do |ss|
     ss.dependency             "React/cxxreact"
     ss.dependency             "React/jsi"
+    ss.dependency             "DoubleConversion"
+    ss.dependency             "glog"
     ss.dependency             "Folly", folly_version
     ss.compiler_flags       = folly_compiler_flags
     ss.source_files         = "ReactCommon/jsiexecutor/jsireact/*.{cpp,h}"
@@ -125,6 +127,8 @@
 
   s.subspec "jsi" do |ss|
     ss.dependency             "Folly", folly_version
+    ss.dependency             "DoubleConversion"
+    ss.dependency             "glog"
     ss.compiler_flags       = folly_compiler_flags
     ss.source_files         = "ReactCommon/jsi/*.{cpp,h}"
     ss.private_header_files = "ReactCommon/jsi/*.h"
@@ -142,6 +146,8 @@
     ss.dependency             "React/jsinspector"
     ss.dependency             "boost-for-react-native", "1.63.0"
     ss.dependency             "Folly", folly_version
+    ss.dependency             "DoubleConversion"
+    ss.dependency             "glog"
     ss.compiler_flags       = folly_compiler_flags
     ss.source_files         = "ReactCommon/cxxreact/*.{cpp,h}"
     ss.exclude_files        = "ReactCommon/cxxreact/SampleCxxModule.*"
